### PR TITLE
Mark other VM imports as unsupported

### PR DESCRIPTION
**Pull request by parlough**
None of these have runtime support on web platforms, similar to `dart:io`.

### CODE REVIEW

**Comment by domesticmouse** [comment]
@kevmoo does this look about right?

**Review by kevmoo — APPROVED** [review]

File: lib/src/flutter_web.dart [hunk 1]
```diff
@@ -43,8 +50,8 @@ class FlutterWebManager {
 
   String getUnsupportedImport(Set<String> imports) {
     for (final import in imports) {
-      // All dart: imports are ok, except dart:io
-      if (import.startsWith('dart:') && import != 'dart:io') {
+      // All non-VM dart: imports are ok
+      if (import.startsWith('dart:') && !_dartVmImports.contains(import)) {
```
Comment (kevmoo) [inline lib/src/flutter_web.dart:54]: although it might be better to do an allow list

Comment (kevmoo) [reply lib/src/flutter_web.dart:54]: image

Comment (domesticmouse) [reply lib/src/flutter_web.dart:54]: @parlough I concur with @kevmoo's suggestion to use a whitelist.

@kevmoo is there a source for the list of supported libraries? It'd be useful to link to this location in the source code.

Comment (kevmoo) [reply lib/src/flutter_web.dart:54]: dartdoc pulls it from somewhere - https://api.dart.dev/stable/2.10.5/index.html – @jcollins-g ?

Comment (kevmoo) [reply lib/src/flutter_web.dart:54]: *cough* allowlist *cough* 😄

Comment (parlough) [reply lib/src/flutter_web.dart]: If we're going with the `allowList` don't we need to include the web libraries as well here? For the HTML version of DartPad at least but even the flutter web version can use them, but a separate `info` is provided suggesting to only use them in packages.

I've gone ahead and included them, but let me know if you'd like to see any changes.

I didn't know how to programmatically retrieve the web-supported libraries, perhaps the analyzer can do so. As for the `dartdoc` functionality that seems to be due to Dartdoc categories, as can be seen here - https://github.com/dart-lang/sdk/blob/master/sdk/lib/mirrors/mirrors.dart#L56

Comment (domesticmouse) [reply lib/src/flutter_web.dart]: Good point, we need to add `dart:html` and friends to the allow list, or a bunch of our samples will die an ignoble death. I believe we should add the following:

- `dart:html`
- `dart:indexed_db`
- `dart:js`
- `dart:js_util`
- `dart:svg`
- `dart:web_gl`
- `dart:web_sql`

Adding @johnpryan as sanity reviewer =)

File: lib/src/flutter_web.dart [hunk 2]
```diff
@@ -24,11 +24,32 @@ class FlutterWebManager {
...
+    'dart:typed_data',
+    'dart:html',
+    'dart:indexed_db',
+    'dart:js',
+    'dart:js_util',
+    'dart:svg',
+    'dart:web_audio',
+    'dart:web_gl',
+    'dart:web_sql',
+    'dart:ui',
```
Comment (domesticmouse) [inline lib/src/flutter_web.dart:50]: Add dart web imports plz

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: lib/src/flutter_web.dart (hunk 2 above)
Comment (parlough) [reply lib/src/flutter_web.dart:50]: I believe they're in the list already, am I missing any?

Comment (domesticmouse) [reply lib/src/flutter_web.dart:50]: Whups, my bad. LGTM

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
TBR @johnpryan

**Pull request merged**